packages/strapi-plugin-upload/controllers/upload/admin.js

```diff
@@ -199,10 +199,9 @@ const findEntityAndCheckPermissions = async (ability, action, model, id) => {
 
   const pm = strapi.admin.services.permission.createPermissionsManager({ ability, action, model });
 
-  const roles = _.has(file, 'created_by.id')
-    ? await strapi.query('role', 'admin').find({ 'users.id': file[CREATED_BY_ATTRIBUTE].id }, [])
-    : [];
-  const fileWithRoles = _.set(_.cloneDeep(file), 'created_by.roles', roles);
+  const author = await strapi.admin.services.user.findOne({ id: file[CREATED_BY_ATTRIBUTE] });
```

Comment:
you can populate only the roles in the findOne

Reply:
Yes, I thought that since we already need the roles, maybe users will find other information useful. I can keep only the roles for the moment though